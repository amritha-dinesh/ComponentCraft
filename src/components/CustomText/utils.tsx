import { TextStyle } from "react-native";
import { colors } from "../../styles/themes/colors";

export const VariantTypesPossible = [
  "headingLarge",
  "headingNormal",
  "headingSmall",
  "heading",
  "subHeadingLarge",
  "subHeadingNormal",
  "subHeadingSmall",
  "subHeading",
  "bodyLarge",
  "bodyNormal",
  "bodySmall",
  "body",
] as const;

export const TextTypesPossible = [
  "primaryLight",
  "primaryNormal",
  "primaryBold",
  "primary",
  "secondaryLight",
  "secondaryNormal",
  "secondaryBold",
  "secondary",
  "successLight",
  "successNormal",
  "successBold",
  "success",
  "errorLight",
  "errorNormal",
  "errorBold",
  "error",
  "warningLight",
  "warningNormal",
  "warningBold",
  "warning",
  "light",
  "normal",
  "bold",
] as const;

export const VariantTypes: {
  [key in (typeof VariantTypesPossible)[number]]: TextStyle;
} = {
  headingLarge: {
    fontSize: 30,
    fontWeight: "bold",
    lineHeight: 36,
    letterSpacing: 0,
  },
  headingNormal: {
    fontSize: 26,
    fontWeight: "bold",
    lineHeight: 32,
    letterSpacing: 0,
  },
  headingSmall: {
    fontSize: 22,
    fontWeight: "bold",
    lineHeight: 28,
    letterSpacing: 0,
  },
  heading: {
    fontSize: 26,
    fontWeight: "bold",
    lineHeight: 32,
    letterSpacing: 0,
  },

  subHeadingLarge: {
    fontSize: 24,
    fontWeight: "normal",
    lineHeight: 28,
    letterSpacing: 0,
  },
  subHeadingNormal: {
    fontSize: 22,
    fontWeight: "normal",
    lineHeight: 26,
    letterSpacing: 0,
  },
  subHeadingSmall: {
    fontSize: 20,
    fontWeight: "normal",
    lineHeight: 24,
    letterSpacing: 0,
  },
  subHeading: {
    fontSize: 22,
    fontWeight: "normal",
    lineHeight: 26,
    letterSpacing: 0,
  },

  bodyLarge: {
    fontSize: 20,
    fontWeight: "normal",
    lineHeight: 26,
    letterSpacing: 0,
  },
  bodyNormal: {
    fontSize: 18,
    fontWeight: "normal",
    lineHeight: 22,
    letterSpacing: 0,
  },
  bodySmall: {
    fontSize: 16,
    fontWeight: "normal",
    lineHeight: 20,
    letterSpacing: 0,
  },
  body: {
    fontSize: 18,
    fontWeight: "normal",
    lineHeight: 22,
    letterSpacing: 0,
  },
};

export const TextTypes: {
  [key in (typeof TextTypesPossible)[number]]: TextStyle;
} = {
  primaryLight: {
    color: colors.blue300,
  },
  primaryNormal: {
    color: colors.blue600,
  },
  primaryBold: {
    color: colors.blue800,
  },
  primary: {
    color: colors.blue600,
  },

  secondaryLight: {
    color: colors.grey500,
  },
  secondaryNormal: {
    color: colors.grey700,
  },
  secondaryBold: {
    color: colors.grey900,
  },
  secondary: {
    color: colors.grey700,
  },

  successLight: {
    color: colors.green300,
  },
  successNormal: {
    color: colors.green600,
  },
  successBold: {
    color: colors.green900,
  },
  success: {
    color: colors.green600,
  },

  errorLight: {
    color: colors.red300,
  },
  errorNormal: {
    color: colors.red600,
  },
  errorBold: {
    color: colors.red900,
  },
  error: {
    color: colors.red600,
  },

  warningLight: {
    color: colors.yellow300,
  },
  warningNormal: {
    color: colors.yellow600,
  },
  warningBold: {
    color: colors.yellow800,
  },
  warning: {
    color: colors.yellow600,
  },

  light: {
    color: colors.grey400,
  },
  normal: {
    color: colors.grey700,
  },
  bold: {
    color: colors.black,
  },
};

export const TextOverflowModeStyles: { [key: string]: any } = {
  clipTheEnd: "clip",
  ignoreTheStart: "head",
  ignoreTheMiddle: "middle",
  ignoreTheEnd: "tail",
};
