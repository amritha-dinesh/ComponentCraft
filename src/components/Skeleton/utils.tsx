import { Animated } from "react-native";

export const generateArray = (dataCount: number) => {
  const data = [];
  for (let i = 1; i <= dataCount; i++) {
    data.push(i);
  }
  return data;
};

export const slideUpStyle = (fadeAnim: Animated.AnimatedValue) => {
  const Style = {
    opacity: fadeAnim,
    transform: [
      {
        translateY: fadeAnim.interpolate({
          inputRange: [0, 1],
          outputRange: [100, 2],
        }),
      },
    ],
  };
  return Style;
};
export const slideDownStyle = (fadeAnim: Animated.AnimatedValue) => {
  const Style = {
    opacity: fadeAnim,
    transform: [
      {
        translateY: fadeAnim.interpolate({
          inputRange: [0, 1],
          outputRange: [-100, 2],
        }),
      },
    ],
  };
  return Style;
};

export const slideLeftStyle = (fadeAnim: Animated.AnimatedValue) => {
  const Style = {
    opacity: fadeAnim,
    transform: [
      {
        translateX: fadeAnim.interpolate({
          inputRange: [0, 1],
          outputRange: [100, 2],
        }),
      },
    ],
  };
  return Style;
};

export const slideRightStyle = (fadeAnim: Animated.AnimatedValue) => {
  const Style = {
    opacity: fadeAnim,
    transform: [
      {
        translateX: fadeAnim.interpolate({
          inputRange: [0, 1],
          outputRange: [-100, 2],
        }),
      },
    ],
  };
  return Style;
};
