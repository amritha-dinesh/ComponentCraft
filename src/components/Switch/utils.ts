import { Platform } from 'react-native';

import {
  grey800,
  grey50,
  white,
} from '../../styles/themes/v2/colors';

type BaseProps = {
  disabled?: boolean;
  value?: boolean;
};

const getCheckedColor = ({
  color,
}: {
  color?: string;
}) => {
  if (color) {
    return color;
  }
};

const getThumbTintColor = ({
  disabled,
  value,
  checkedColor,
}: BaseProps & { checkedColor?: string }) => {
  const isIOS = Platform.OS === 'ios';

  if (isIOS) {
    return undefined;
  }

  if (disabled) {
    return grey800;
  }

  if (value) {
    return checkedColor;
  }
  return grey50;
};

const getOnTintColor = ({
  disabled,
  value,
  checkedColor,
}: BaseProps & { checkedColor?: string }) => {
  const isIOS = Platform.OS === 'ios';

  if (isIOS) {
    return checkedColor;
  }

  if (disabled) {
    return white;
  }

  if (value) {
    return checkedColor;
  }
  return 'rgb(178, 175, 177)';
};

export const getSwitchColor = ({
  disabled,
  value,
  color,
}: BaseProps & { color?: string }) => {
  const checkedColor = getCheckedColor({color});

  return {
    onTintColor: getOnTintColor({ disabled, value, checkedColor }),
    thumbTintColor: getThumbTintColor({ disabled, value, checkedColor }),
    checkedColor,
  };
};