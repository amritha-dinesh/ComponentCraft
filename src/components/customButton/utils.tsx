export type ButtonMode = "text" | "contained" | "outlined";

type BaseProps = {
  isMode: (mode: ButtonMode) => boolean;
  disabled?: boolean;
  color?: string;
  buttonColor?: string;
  disabledButtonColor?: string;
  outlinedColor?: string;
  borderColor?: string;
};
export const getButtonColors = ({
  mode,
  buttonColor,
  color,
  disabled,
  disabledButtonColor,
}: {
  mode: ButtonMode;
  buttonColor?: string;
  color?: string;
  disabled?: boolean;
  disabledButtonColor?: string;
  outlinedColor?: string;
}) => {
  const isMode = (modeToCompare: ButtonMode) => mode === modeToCompare;

  const getButtonBackgroundColor = ({
    disabled,
    buttonColor,
    isMode,
    disabledButtonColor,
  }: BaseProps) => {
    if (isMode("outlined") || isMode("text")) {
      return "transparent";
    }
    if (buttonColor && !disabled) {
      return buttonColor;
    }
    if (isMode("contained") && disabledButtonColor && disabled) {
      return disabledButtonColor;
    }
  };

  const backgroundColor = getButtonBackgroundColor({
    isMode,
    disabled,
    buttonColor,
    disabledButtonColor,
  });

  const getButtonTextColor = ({
    disabled,
    color,
  }: Omit<BaseProps, "isMode">) => {
    if (color && !disabled) {
      return color;
    } else {
      return null;
    }
  };
  const textColor = getButtonTextColor({
    disabled,
    color,
  });
  const getButtonBorderColor = ({
    isMode,
    disabled,
    borderColor,
  }: BaseProps) => {
    if (disabled && isMode("text")) {
      return "transparent";
    }
    if (isMode("outlined") && isMode("contained")) {
      return borderColor;
    }
  };
  const borderColor = getButtonBorderColor({ isMode, disabled });
  const getButtonBorderWidth = ({ isMode }: Omit<BaseProps, "disabled">) => {
    if (isMode("text")) {
      return 0;
    } else {
      return 1;
    }
  };

  const borderWidth = getButtonBorderWidth({ isMode });
  const getButtonBorderRadius = ({ isMode }: Omit<BaseProps, "disabled">) => {
    if (isMode("text")) {
      return 0;
    } else {
      return 5;
    }
  };

  const borderRadius = getButtonBorderRadius({ isMode });

  return {
    backgroundColor,
    borderColor,
    textColor,
    borderWidth,
    borderRadius,
  };
};
