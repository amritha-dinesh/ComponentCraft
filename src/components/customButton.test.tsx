import * as React from "react";
import { render } from "@testing-library/react-native";
import CustomButton from "./customButton";
import { getButtonColors } from "./utils";

it("renders text button by default", () => {
  const tree = render(
    <CustomButton buttonTitle="helloworld"></CustomButton>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders text button with mode", () => {
  const tree = render(
    <CustomButton buttonTitle="helloworld" mode="text"></CustomButton>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders contained button with mode", () => {
  const tree = render(
    <CustomButton mode="contained" buttonTitle="helloworld"></CustomButton>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders disabled button", () => {
  const tree = render(
    <CustomButton disabled buttonTitle="helloworld"></CustomButton>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
it("renders button with custom testID", () => {
  const tree = render(
    <CustomButton
      testID={"custom:testID"}
      buttonTitle="helloworld"
    ></CustomButton>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders button with an accessibility label", () => {
  const tree = render(
    <CustomButton accessibilityLabel={"label"}></CustomButton>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders button with an accessibility hint", () => {
  const tree = render(
    <CustomButton accessibilityHint={"hint"}></CustomButton>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
