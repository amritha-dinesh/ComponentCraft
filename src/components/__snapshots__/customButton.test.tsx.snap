// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with an accessibility hint 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityHint="hint"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="button-text"
    />
  </View>
</View>
`;

exports[`renders button with an accessibility label 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityLabel="label"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="button-text"
    />
  </View>
</View>
`;

exports[`renders button with custom testID 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="custom:testID"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="custom:testID-text"
    >
      helloworld
    </Text>
  </View>
</View>
`;

exports[`renders contained button with mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="button-text"
    >
      helloworld
    </Text>
  </View>
</View>
`;

exports[`renders disabled button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="button-text"
    >
      helloworld
    </Text>
  </View>
</View>
`;

exports[`renders text button by default 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="button-text"
    >
      helloworld
    </Text>
  </View>
</View>
`;

exports[`renders text button with mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          },
          {
            "color": "#000000",
          },
        ]
      }
      testID="button-text"
    >
      helloworld
    </Text>
  </View>
</View>
`;
